.github/workflows/build.yml

```diff
@@ -0,0 +1,40 @@
+name: Build PyPI
+on:
+  workflow_call:
+  workflow_dispatch:
+  push:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+jobs:
+  linting:
+    name: Reuse linting job
+    uses: ./.github/workflows/lint.yml
+
+  build:
+    needs: linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out source repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 #full history
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      - name: Install dependencies
+        run: pip install poetry pytest reqstool-python-decorators
+      - name: Run unit and integrations tests
+        run: poetry run pytest tests
```

Comment:
In our hatch projects we use: 

`run: hatch run test:pytest --cov=reqstool-python-decorators --cov-report=xml --cov-report=html`

Reply:
yes, but we cant run Poetry with Hatch. And the build kept failing if i tried to do 

```
poetry run pytest --junitxml=build/junit.xml --cov --cov-report=xml:build/coverage.xml tests
```

The command runs fine locally, but fails in the workflow. Since the only test there is does nothing else than create a folder, I figured that I cycle back to this when we actually hase something to test. :P